redwood-treehouse-host/src/commonTest/kotlin/app/cash/redwood/treehouse/TreehouseAppContentTest.kt

```diff
@@ -20,35 +20,59 @@ import assertk.assertThat
 import assertk.assertions.isEmpty
 import assertk.assertions.isEqualTo
 import assertk.assertions.isNotEmpty
+import kotlin.coroutines.EmptyCoroutineContext
 import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
 import kotlin.test.Test
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.runBlocking
 import kotlinx.coroutines.test.UnconfinedTestDispatcher
 import kotlinx.coroutines.test.runTest
 
+/**
+ * This test focuses on how [TreehouseAppContent] behaves in response to lifecycle events from the
+ * code ([CodeSession]), the UI ([TreehouseView]), and the content ([ZiplineTreehouseUi]).
+ */
 @OptIn(ExperimentalCoroutinesApi::class)
 class TreehouseAppContentTest {
   private val eventLog = EventLog()
+  private val appScope = CoroutineScope(EmptyCoroutineContext)
 
   private val dispatcher = UnconfinedTestDispatcher()
   private val eventPublisher = FakeEventPublisher()
-  private val codeHost = FakeCodeHost(eventLog, eventPublisher)
   private val dispatchers = FakeDispatchers(dispatcher, dispatcher)
+  private val onBackPressedDispatcher = FakeOnBackPressedDispatcher(eventLog)
+  private val codeHost = FakeCodeHost(
+    eventLog = eventLog,
+    eventPublisher = eventPublisher,
+    dispatchers = dispatchers,
+    appScope = appScope,
+    frameClockFactory = FakeFrameClock.Factory,
+  )
   private val codeListener = FakeCodeListener(eventLog)
   private val uiConfiguration = UiConfiguration()
 
+  @BeforeTest
+  fun setUp() {
+    runBlocking {
+      codeHost.start()
+      eventLog.takeEvent("codeHost.collectCodeUpdates()")
```

Comment:
I added an event for this so I can tell when we’re receiving new code and when we’re not. In this test it’s not that interesting!